src/material/icon/icon-registry.ts

```diff
@@ -73,9 +73,9 @@ class SvgIconConfig {
   url: SafeResourceUrl | null;
   svgElement: SVGElement | null;
 
-  constructor(url: SafeResourceUrl);
-  constructor(svgElement: SVGElement);
-  constructor(data: SafeResourceUrl | SVGElement) {
+  constructor(url: SafeResourceUrl, viewBox?: string);
```

Comment:
Instead of adding an argument for viewBox specifically, can we just add an options object? This would capture e.g.  `preserveAspectRatio`